### ISSUE TITLE
initializing hyperparamters of GP regression + Delta 

### ISSUE THREAD

**Issue by hyunjae-woo** [firasmidani/amiga#14]
Hi, I'm using amiga to estimate the growth rate for high-throughput microplate reader. I have little knowledge about the technical part of regression and therefore have a question about the different mode type for initializing hyperparamters of GP regression which are  "default", "moments-based-proper" or "moments-based-fast". Would moments-based be more accurate than default? Would there be significant difference between them?

Also for handling nonpositive, if I am using the microplate reader with 10 minutes interval for 24 hours, would the default Delta number, which is 5, be sufficient? Thank you!

**Comment by firasmidani** [comment]
Good questions. 

Regarding __hyperparameters initialization__: I implemented these different choices to help a researcher who was trying to use AMiGA for modelling viral growth curves. Unlike microbial growth curves, their viral growth curves had a y-axis based on fluorescence which ranged from 0 to the 10,000s ;  for examples, see [preprint](https://www.biorxiv.org/content/10.1101/2022.06.21.496950v1.abstract). However, AMiGA was optimized for modelling optical density curves with small values (range of 0 to 5). This wide range of values prevented the Gaussian Process regression from reaching the optimal fit using the default hyperparameters. So, I updated AMiGA to include these additional choices that you mention. Difference between "moments-based-proper" and "moments-based-fast": the former is more accurate but slower, the latter is faster but less accurate. The fit of "optical density" is pretty good by both, but the growth rates are mostly accurate with the 'proper' method. If you are using AMiGA to model optical density, I would suggest that you simply use the "default" option. If you are using AMiGA to model a different growth estimate (say fluorescence) with very wide range, I suggest that you try all three options and look at the plot of the derivatives (use ```--plot-derivative``` argument) to compare results and see which one gives you a better estimate of growth curves. 

Regarding __Delta__: The Delta should be optimized for the growth characteristics of your bacteria of choice. Personally, I found a Delta value of 5 is good enough for C. difficile because it has short lag phases and the optical density stays close to the initial value over the first hour. After all, a delta of 5 (with interval value of 10 minutes) indicates that the *offset* is computed using the variation in the optical density in the first 50 minutes (For more info, see [documentation](https://firasmidani.github.io/amiga/doc/configuration.html)).

I would caution on using large Delta value. Let's say you haver microbial growth with very short lag time and your Delta is say 18 (for intervals of 10 minutes, that's equivalent to 3 hours). Then, the *offset* is computed as the median variation in the optical density over the first three hours. If these three hours includes dramatic growth during the exponential stage, then the offset will be much larger than you would want it.
